addons/l10n_tr_nilvera/models/res_partner.py

```diff
@@ -16,49 +17,75 @@ class ResPartner(models.Model):
     invoice_edi_format = fields.Selection(selection_add=[('ubl_tr', "UBL TR 1.2")])
     l10n_tr_nilvera_customer_status = fields.Selection(
         selection=[
-            ('not_checked', "Not Checked"),
+            ('not_checked', "Not Verified"),
             ('earchive', "E-Archive"),
             ('einvoice', "E-Invoice"),
         ],
         string="Nilvera Status",
-        compute='_compute_nilvera_customer_status_and_alias_id',
-        store=True,
         copy=False,
         default='not_checked',
+        readonly=True,
+        tracking=True,
     )
     l10n_tr_nilvera_customer_alias_id = fields.Many2one(
         comodel_name='l10n_tr.nilvera.alias',
         string="Alias",
-        compute='_compute_nilvera_customer_status_and_alias_id',
+        compute='_compute_nilvera_customer_alias_id',
         domain="[('partner_id', '=', id)]",
         copy=False,
         store=True,
         readonly=False,
     )
 
-    # This field is only used technically for optimisation purposes. It's needed for check_nilvera_customer.
+    # This field is only used technically for optimisation purposes. It's needed for _check_nilvera_customer.
     l10n_tr_nilvera_customer_alias_ids = fields.One2many(
         comodel_name='l10n_tr.nilvera.alias',
         inverse_name="partner_id",
     )
 
-    @api.depends('vat', 'invoice_edi_format')
-    def _compute_nilvera_customer_status_and_alias_id(self):
-        if modules.module.current_test:
+    @api.depends('l10n_tr_nilvera_customer_alias_ids')
+    def _compute_nilvera_customer_alias_id(self):
+        for record in self:
+            record.l10n_tr_nilvera_customer_alias_id = record.l10n_tr_nilvera_customer_alias_ids[:1]
+
+    def _send_user_notification(self, type, message, action_button=None):
+        notification_type = 'account_notification' if action_button else 'simple_notification'
+        self.env.user._bus_send(notification_type, {
+            'type': type,
+            'message': message,
+            'action_button': action_button,
+        })
+
+    def l10n_tr_check_nilvera_customer(self):
+        if not self:
             return
-        for partner in self:
-            if partner.vat and partner.invoice_edi_format == 'ubl_tr':
-                try:
-                    partner.check_nilvera_customer()
-                except UserError:
-                    # In case of an internet connection issue, exit silently.
-                    continue
-            else:
-                # Reset the alias if no VAT or UBL format changed.
-                partner.l10n_tr_nilvera_customer_status = 'not_checked'
-                partner.l10n_tr_nilvera_customer_alias_id = False
 
-    def check_nilvera_customer(self):
+        results = defaultdict(lambda: self.env['res.partner'])
+        for record in self.filtered("vat"):
```

Comment:
to avoid doing a filtered that will cost performance you could do an early break